rclcpp/include/rclcpp/intra_process_manager_impl.hpp

```diff
@@ -175,7 +189,8 @@ class IntraProcessManagerImpl : public IntraProcessManagerImplBase
     }
 
     // Figure out what subscriptions should receive the message.
-    auto & destined_subscriptions = subscription_ids_by_topic_[publisher->get_topic_name()];
+    auto & destined_subscriptions =
+      subscription_ids_by_topic_[FixedSizeString(publisher->get_topic_name())];
```

Comment:
I never noticed this before, and it's out of scope for this pr, but this is a really expensive operation to do in `store_intra_process_message`. I'll have to think about a better way to do this that doesn't require a string compare based search in the common case.

Reply:
Sorry to disrupt, but what about a `std::unordered_map` i.e. a hash map?

Reply:
I was thinking more along the lines of storing this information so that it can be accessed by id (integer), as it only changes when something is added or removed. An `unordered_map` would be an improvement on the existing approach, however.

Reply:
I agree that  comparing strings is too expensive. I will only solve the bug in this PR.

Reply:
> I agree that comparing strings is too expensive. I will only solve the bug in this PR.

👍